concourse/scripts/bosh-cli.sh

```diff
@@ -10,12 +10,17 @@ export TARGET_CONCOURSE=deployer
 $("${SCRIPT_DIR}/environment.sh")
 "${SCRIPT_DIR}/fly_sync_and_login.sh"
 
-OUTPUT_FILE=$(mktemp -t bosh-cli.XXXXXX)
-trap 'rm -f "${OUTPUT_FILE}"' EXIT
+BUILD_NUMBER=$($FLY_CMD -t "${FLY_TARGET}" trigger-job -j create-cloudfoundry/bosh-cli | \
+	awk '/started create-cloudfoundry\/bosh-cli/ { print $3 }' | \
+	tr -d '#')
 
-$FLY_CMD -t "${FLY_TARGET}" trigger-job -j create-cloudfoundry/bosh-cli -w | tee "${OUTPUT_FILE}"
-
-BUILD_NUMBER=$(awk '/started create-cloudfoundry\/bosh-cli/ { print $3 }' "${OUTPUT_FILE}" | tr -d '#')
+while $FLY_CMD -t "${FLY_TARGET}" builds -j create-cloudfoundry/bosh-cli | awk '{print $3 $4}' | grep "${BUILD_NUMBER}pending" >/dev/null; do
+	echo "waiting for create-cloudfoundry/bosh-cli container (${BUILD_NUMBER}) to start..."
+	sleep 2
+done
+sleep 8 # required since even after status is 'started' it takes a few seconds
 
+echo "hijacking create-cloudfoundry/bosh-cli container (${BUILD_NUMBER})..."
+trap '$FLY_CMD -t "${FLY_TARGET}" abort-build -j create-cloudfoundry/bosh-cli -b "${BUILD_NUMBER}"' EXIT
```

Comment:
I've discovered that this has a problem on persistent environments. We've configured a 5 minute idle timeout there, so if the bosh-cli session is longer than that, this fails with an authentication error.

Reply:
Damn.

This is the auth timeout right? ...not the [concourse step timeout](https://concourse.ci/timeout-step.html) which could be altered?

Reply:
Correct, this is the auth timeout.

This isn't a massive problem because the while loop in the task itself will cause the task to exit in up to 5 mins. It is a bit of noise though.